packages/flutter_tools/lib/src/commands/analyze_base.dart

```diff
@@ -103,6 +115,41 @@ abstract class AnalyzeBase {
 
     return dartDocMessage;
   }
+
+  /// Print an analysis summary.
+  static String generateErrorsMessage({
+    int issueCount,
+    int issueDiff,
+    int files,
+    String seconds,
+    int undocumentedMembers,
+    String dartDocMessage,
+  }) {
```

Comment:
This code is less clear than what it is replacing. Consider:

* Keep `if/else` branching as flat as possible, so it is easier to follow the conditions.
* Use a StringBuffer to build up intermediary strings.
* use required on parameters that need to be passed, or else provide them with default values (unless that default value can be null)